extensions/wikia/Insights/specials/TemplatesWithoutTypePage.class.php

```diff
@@ -0,0 +1,175 @@
+
+	function __construct( $name = self::TEMPLATES_WITHOUT_TYPE_TYPE ) {
+		parent::__construct( $name );
+	}
+
+	public function isListed() {
+		return false;
+	}
+
+	public function sortDescending() {
+		return true;
+	}
+
+	public function isExpensive() {
+		return true;
+	}
+
+	/**
+	 * A wrapper for calling the querycache table
+	 *
+	 * @param bool $offset
+	 * @param int $limit
+	 * @return ResultWrapper
+	 */
+	public function doQuery( $offset = false, $limit = self::LIMIT ) {
+		return $this->fetchFromCache( $limit, $offset );
+	}
+
+	/**
+	 * Update the querycache table
+	 *
+	 * @param bool $limit Only for consistency
+	 * @param bool $ignoreErrors Only for consistency
+	 * @return bool|int
+	 */
+	public function recache( $limit = false, $ignoreErrors = true ) {
+		$dbw = wfGetDB( DB_MASTER );
+
+		/**
+		 * 1. Get the new data first
+		 */
+		$templatesWithoutType = $this->reallyDoQuery();
+		$dbw->begin();
+
+		/**
+		 * 2. Delete the existing records
+		 */
+		( new WikiaSQL() )
+			->DELETE( 'querycache' )
+			->WHERE( 'qc_type' )->EQUAL_TO( $this->getName() )
+			->run( $dbw );
+
+		/**
+		 * 3. Insert the new records if the $templatesWithoutType array is not empty
+		 */
+		$num = 0;
+		if ( !empty( $templatesWithoutType ) ) {
+
+			( new WikiaSQL() )
+				->INSERT()->INTO( 'querycache', [
+					'qc_type',
+					'qc_value',
+					'qc_namespace',
+					'qc_title'
+				] )
+				->VALUES( $templatesWithoutType )
+				->run( $dbw );
+
+			$num = $dbw->affectedRows();
+			if ( $num === 0 ) {
+				$dbw->rollback();
+				$num = false;
+			} else {
+				$dbw->commit();
+			}
+		}
+
+		return $num;
+	}
+
+	/**
+	 * Prepares a list of not recognized templates that are used on content namespace pages on a wikia.
+	 *
+	 * return array structure
+	 * [
+	 *   [ {qc_type}, {qc_value}, {qc_ns}, {qc_title} ],
+	 *   ...
+	 * ]
+	 * @param bool $limit Only for consistency
+	 * @param bool $offset Only for consistency
+	 * @return array
+	 */
+	public function reallyDoQuery( $limit = false, $offset = false ) {
+		global $wgCityId, $wgContentNamespaces;
+		$dbr = wfGetDB( DB_SLAVE, [ $this->getName(), __METHOD__, 'vslow' ] );
+		$templatesWithoutType = [];
+
+		$CNTemplatesOnWikia = $this->getContentNamespacesTemplates( $dbr, $wgContentNamespaces );
+		$recognizedTemplates  = $this->getNotRecognizedTemplatesOnWikia(
+			( new TemplateClassificationService() ),
+			$wgCityId
+		);
+
+		$CNRecognizedTemplates = $this->intersectSets( $CNTemplatesOnWikia, $recognizedTemplates );
+
+		foreach( $CNRecognizedTemplates as $recognizedTemplate ) {
+			$title = Title::newFromID( $recognizedTemplate );
+			$links = $title->getIndirectLinks();
+				$templatesWithoutType[] =
+				[
+					$this->getName(),
+					count( $links ),
+					NS_TEMPLATE,
+					$title->getDBkey()
+				];
+
+		}
+
+		return $templatesWithoutType;
+	}
+
+	private function getContentNamespacesTemplates($db, $contentNamespaces) {
+		$sql = ( new \WikiaSQL() )
```

Comment:
Actually... Why are you doing this? The acceptance criteria say nothing about focusing only on the Templates included in the content namespace.